delete.go

```diff
@@ -58,24 +59,28 @@ status of "ubuntu01" as "stopped" the following will delete resources held for
 			}
 			return err
 		}
-		// When --force is given, we kill all container processes and
-		// then destroy the container. This is done even for a stopped
-		// container, because (in case it does not have its own PID
-		// namespace) there may be some leftover processes in the
-		// container's cgroup.
-		if force {
-			return killContainer(container)
-		}
+
 		s, err := container.Status()
 		if err != nil {
 			return err
 		}
 		switch s {
 		case libcontainer.Stopped:
+			// For a stopped container, because (in case it does not have
+			// its own PID namespace) there may be some leftover processes
+			// in the container's cgroup.
+			if !container.Config().Namespaces.IsPrivate(configs.NEWPID) {
+				return killContainer(container)
+			}
 			return container.Destroy()
```

Comment:
Maybe drop `killContainer` entirely here and just call `container.Kill()` and then `container.Destroy()`? The code is more readable this way.

Or, rename killContainer to killAndDestroy.

Reply:
> Maybe drop `killContainer` entirely here and just call `container.Kill()` and then `container.Destroy()`? The code is more readable this way.

Good idea, it is indeed more readable than before!